.github/workflows/codeql-analysis.yml

```diff
@@ -9,36 +9,115 @@ name: CodeQL security analysis
 
 on:
   workflow_call:
-
 jobs:
+  analyzeExcludingWarnings:
+    name: Analyze add-on excluding warnings
+    runs-on: windows-latest
+    permissions:
+      actions: read
+      contents: write
+      security-events: write
+      issues: write
+      pull-requests: write
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ 3.11 ]
+        language: ['javascript', 'python']
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://git.io/codeql-language-support
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Download add-on
+        uses: actions/download-artifact@v4
+        with:
+          name: addon
+      - name: Unzip add-on
+        run: |
+          Copy-Item addon.nvda-addon addon.zip
+          Expand-Archive addon.zip
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: ${{ matrix.language }}
+          config-file: ./.github/codeql/codeql-config-excluding-warnings.yml
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v3
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v3
+        with:
+          output: results
+      - name: Download add-on metadata
+        uses: actions/download-artifact@v4
+        with:
+          name: addonMetadata
+      - name: Set security analysis status
+        id: setSecurityAnalysisStatus
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const setSecurityAnalysisStatus = require('./.github/workflows/securityAnalysis.js')
+            const path = 'results/${{ matrix.language }}.sarif'
+            setSecurityAnalysisStatus({core}, path)
+      - name: Upload results
+        id: uploadResults
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: results-${{ matrix.language }}
+          path: results/*.sarif
+          overwrite: true
+      - name: Create pull request
+        id: cpr
+        if: failure()
+        uses: peter-evans/create-pull-request@v6
+        with:
+          add-paths: reviewedAddons.json
+          title: Add reviewed add-on (${{ github.event.issue.title }})
```

Comment:
can this use the add-on id? the issue title often doesn't match

Reply:
Yes, I think it would be better.